tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -187,12 +175,23 @@ sub find_sriov_ethernet_devices {
     my @sriov_devices;
     foreach (@nic_devices) {
         if ((script_run "lspci -v -s $_ | grep -q 'SR-IOV'") == 0) {
-            #only those vfs whose pv can be brought up can be passed through to guest vms
+            # Any VFs can be passed through to guests
+            # But only those VFs whose pv has physical network connection can get an IP from DHCP server
             my $nic = script_output "ls -l /sys/class/net |grep $_ | awk '{print \$9}'";
-            script_run "echo \"BOOTPROTO='none'\" > /etc/sysconfig/network/ifcfg-$nic" unless $nic eq get_var('SUT_NETDEVICE', 'eth0');
-            push @sriov_devices, $_ if (script_run("ifup $nic") == 0);
+            if ($nic eq get_var('SUT_NETDEVICE', 'eth0')) {
+                push @sriov_devices, $_;
+            }
+            else {
+                assert_script_run("ip link set $nic up");
+                if (script_output("cat /sys/class/net/$nic/carrier") eq '1') {
```

Comment:
Wow~, this goes more and more professional for network now

Reply:
Yes, I spent some time in figuring out if there is a physical connection to a network card.